### ISSUE TITLE
Button action triggered before animation completes

### ISSUE THREAD

**Issue by ryryan** [Orderella/PopupDialog#4]
When using the below code and selecting one of the buttons which triggers a segue, there is a delay moving views. This is possibly due to the PopupDialog not dismissing quick enough.

A dismiss/hide method would be handy for many use cases.

```
let title = "Account Required"
let message = "Would you like to create an account or sign in?"

 let alert = PopupDialog(title: title, message: message)

let createAccount = DefaultButton(title: "Create Account") {
    self.performSegueWithIdentifier("createAccount", sender: self)
}

let signIn = DefaultButton(title: "Sign In") {         
    self.performSegueWithIdentifier("signIn", sender: self)
 }

let cancel = CancelButton(title: "Cancel") {
     print("Dismiss")
}

alert.addButtons([createAccount, signIn, cancel])

self.presentViewController(alert, animated: true, completion: nil)
```

**Comment by ryryan** [comment]
Calling the below code works, but it would be a better implementing to have `alert.dismiss`.
`self.dismissViewControllerAnimated(true, completion: nil)`

**Comment by mwfire** [comment]
@ryhart18  Good point. As a matter of fact, I think the button action should only be triggered once the animation completes to prevent these kinds of glitches. I have pushed a fix c137620 to the hotfix branch, it would be great if you gave it a go. Let me know if that works for you.

In order to install the version with the fix, use the following branch in your podfile:

`pod 'PopupDialog', :git => 'https://github.com/Orderella/PopupDialog.git', :branch => 'hotfix/deferredButtonAction'`

**Comment by mwfire** [comment]
This has been released with V0.1.6, 5017566.
